components/scream/src/physics/shoc/tests/shoc_varorcovar_tests.cpp

```diff
@@ -0,0 +1,270 @@
+  //   to large values to make sure they are not modified
+  Real varorcovar[nlevi] = {100., 0., 0., 0., 100.};
+  // Tuning factor
+  Real tunefac=1.0;  
+
+  // Initialzie data structure for bridgeing to F90
+  SHOCVarorcovarData SDS(shcol, nlev, nlevi, tunefac);
+
+  // Test that the inputs are reasonable.
+  REQUIRE(SDS.nlevi - SDS.nlev == 1);
+  REQUIRE(SDS.shcol > 0);
+  
+  // Fill in test data 
+  for(Int s = 0; s < SDS.shcol; ++s) {
+    // First on the nlev grid
+    for(Int n = 0; n < SDS.nlev; ++n) {
+      const auto offset = n + s * SDS.nlev;
+      
+      // Fill invar1 and invar2 with the SAME 
+      //  variable for this case to test the 
+      //  variance calculation of this function
+      SDS.invar1[offset] = invar_theta[n]; 
+      SDS.invar2[offset] = invar_theta[n];
+    }
+    
+    // Now for data on the nlevi grid
+    for(Int n = 0; n < SDS.nlevi; ++n) {
+      const auto offset = n + s * SDS.nlevi;
+      
+      SDS.tkh_zi[offset] = tkh_zi[n];
+      SDS.isotropy_zi[offset] = isotropy_zi[n];
+      SDS.dz_zi[offset] = dz_zi[n];
+      SDS.varorcovar[offset] = varorcovar[n];
+    }
+  }
+
+  // Check that the inputs make sense
+  
+  REQUIRE(SDS.tunefac > 0.0);
+  // Check to make sure that dz_zi, tkh_zi, and isotropy_zi
+  //  (outside of the boundaries) are greater than zero 
+  for(Int s = 0; s < SDS.shcol; ++s) {
+    // do NOT check boundaries!
+    for(Int n = 1; n < SDS.nlevi-1; ++n) {
+      const auto offset = n + s * SDS.nlevi;
+      REQUIRE(SDS.dz_zi[offset] > 0.0);
+      REQUIRE(SDS.tkh_zi[offset] > 0.0);
+      REQUIRE(SDS.isotropy_zi[offset] > 0.0);
+    }
+    // For this test make sure that invar1 = invar2
+    for(Int n = 0; n < SDS.nlev; ++n){
+      const auto offset = n + s * SDS.nlev;
+      REQUIRE(SDS.invar1[offset] == SDS.invar2[offset]);
+    }
+  }
+  
+  // Call the fortran implementation for variance
+  calc_shoc_varorcovar(nlev, SDS);
+  
+  // Check the results
+  for(Int s = 0; s < SDS.shcol; ++s) {   
+    for(Int n = 0; n < SDS.nlevi; ++n) {
+      const auto offset = n + s * SDS.nlevi;
+
+      // validate that the boundary points have NOT been modified      
+      if (n == 0 || n == nlevi){
+        REQUIRE(SDS.varorcovar[offset] == 100.0);
+      }
+      else{
+      
+      // Validate that all values are greater to 
+      //   or equal to zero 
+      
+        REQUIRE(SDS.varorcovar[offset] >= 0.0);
+      
+        // well mixed layer test
+        if ((invar_theta[n-1] - invar_theta[n]) == 0.0){
+          REQUIRE(SDS.varorcovar[offset] == 0.0);
+        }
+      }
+    } 
+  }
+
+  //********************************************
+  // TEST TWO
+  // Verify the Covariance calculation is valid
+  
+  // Now we check the covariance calculation
+  // Define an array to be total water [g/kg]
+  Real invar_qw[nlev] = {5.0, 10.0, 15.0, 20.0};
+  
+  // NOTE: all other inputs are reused from test one
+  
+  // convert to [kg/kg]
+  invar_qw[:] = invar_qw[:]/1000.0;
```

Comment:
The code is failing to compile on White. I think changing the above to a for loop might help.

Reply:
Thanks @singhbalwinder, that solved the fail on White.... but now tests are failing on Blake (which were previously passing).  I'm having a hard time understanding what the problem is from the console output though...